### PR TITLE
Add nested objects support

### PR DESCRIPTION
**Pull request by Sergej-Popov**
Adds support for nested objects, this allows to create a schema for arbitrary dictionary objects where the nested structure is defined but keys are not. e.g OpenAPI spec

```js

  "paths": {
    "/my-resource": {
      "post": {
        "tags": [
          "resource"
        ],
        "summary": "Create my Resource"

```
In the above example, resource structure is well defined, but the path identifiers are not.

The second commit in this PR adds support for double nested dictionaries. e.g:
```js
{
      NestedDictionaryProp: {
        "KeyA": {
          "KeyA1": {
            DictionaryChildNumberProp: 500
          }
        }
      }
    }
```

## Type of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](/justeat/ts-jsonschema-builder/pullspulls) for the same update/change?
* [x] I have updated the documentation accordingly.

### New Feature Submissions:

* [x] Does your submission pass tests?
* [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -19,4 +18,4 @@
 
 * [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
 * [ ] Have you written new tests for your core changes, as applicable?
-* [ ] Have you successfully ran tests with your changes locally?
\ No newline at end of file
+* [ ] Have you successfully ran tests with your changes locally?
```
Comment (bingnz) [inline .github/pull_request_template.md]:
```suggestion
* [ ] Have you successfully run tests with your changes locally?
```

File: tests/object.spec.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          DictionaryChildBoolProp: null
+        }
+      }
+    };
+
+    const schema = new Schema<Model>()
+      .with(m => m.DictionaryProp,
+        new Schema<DictionaryPropModel>()
+          .with(x => x.DictionaryChildBoolProp, new BooleanSchema({
+            required: true
```
Comment (bingnz) [inline tests/object.spec.ts]: Should this be `required: true` by default?

File: tests/object.spec.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
+import { describe, it } from "mocha";
+
+import { Schema, ArraySchema, StringSchema, AnyOf, NumberSchema, AllOf, BooleanSchema, OneOf } from "../src";
+import { Model, DictionaryPropModel, NestedDictionaryPropModel } from "./models";
+import { assertValid, assertInvalid } from "./assertion";
+
+describe("Dictionary", () => {
+
+  it("Should pass when nested object property satisfies schema ", () => {
```
Comment (bingnz) [inline tests/object.spec.ts]: minor: you have trailing spaces on all of these test names.

**Comment by Sergej-Popov** [comment]
I like static factories idea. I will create an issue to track that separately.

**Pull request merged**